### PR TITLE
Add JSVPSA install tasks

### PR DESCRIPTION
**Pull request by jbickar**

### CODE REVIEW

File: Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    $all_links = null;
+    $cache = null;
+    $drupal_static_cache = null;
+    // Call flush all caches so that the blocks table gets populated.
+    drupal_static_reset();
+    drupal_flush_all_caches();
+    // Revert dem all!
+    // features_rebuild();
+    // features_revert();
+    features_revert('stanford_video');
```
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php]: I wonder if this is still needed. Do we have a ticket to check this?

File: Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    // Small fix for a menu bug
+    db_update('menu_links')
+      ->fields(array(
+        'router_path' => 'admin/manage'
+      ))
+      ->condition('link_path', 'admin/manage')
+      ->execute();
+    // Add some block titles.
+    // module, delta, title
+    $blocks = array(
```
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php:45]: This seems like it should belong in the content section but I guess that it depends on the cache clears. Good here for now unless you know better.

File: Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php (hunk 1 above)
Comment (jbickar) [inline Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php]: Are you talking about just the `stanford_video` Features revert, or all the cache flushing?

File: Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuItems.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    );
+    // For Staff.
+    $items['private/for-staff'] = array(
+      'link_path' => drupal_get_normal_path('private/for-staff'),
+      'link_title' => 'For Staff',
+      'menu_name' => 'menu-menu-private-pages',
+      'weight' => -3,
+    );
+    // Loop through each of the items and save them.
+    foreach ($items as $k => $v) {
```
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuItems.php:167]: This function would be a good candidate for a utility class.

File: Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuPositionRules.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      'link_title' => 'People',
+      'admin_title' => 'People by path',
+      'conditions' => array(
+        'pages' => array(
+          'pages' => 'people/*',
+        ),
+      ),
+    );
+    foreach ($rules as $mp_rule) {
+      $rule = new \Stanford\Utility\Install\InsertMenuRule();
```
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuPositionRules.php:86]: Niiiice.

Comment (jbickar) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuPositionRules.php:86]: Thanks. Trying to bone up on OO.

File: Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuRedirects.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+   *   Installation arguments.
+   */
+  public function execute(&$args = array()) {
+
+    // Create redirects.
+    $redirects = array(
+      'news' => 'news/announcements',
+      'events' => 'events/upcoming-events',
+    );
+    foreach ($redirects as $source => $dest) {
```
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuRedirects.php:26]: This is a good candidate for a utility class.

File: Stanford/Utility/Install/NodeAccessRebuild.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @file
+ * This function removes any view that has been saved to the database.
```
Comment (sherakama) [inline Stanford/Utility/Install/NodeAccessRebuild.php]: Copy pasta is yummy.

File: Stanford/Utility/Install/InsertMenuRule.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @file
+ * This function saves all of the available views in to the database so that
```
Comment (sherakama) [inline Stanford/Utility/Install/InsertMenuRule.php]: mmmm copy pasta.

File: Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuItems.php (hunk 3 above)
Comment (jbickar) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuItems.php:167]: Added as BASIC-1918

File: Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php (hunk 1 above)
Comment (sherakama) [inline Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php]: The video line.

File: Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuRedirects.php (hunk 5 above)
Comment (jbickar) [inline Stanford/JumpstartVPSA/Install/Menu/JSVPSAMenuRedirects.php:26]: BASIC-1919

File: Stanford/Utility/Install/NodeAccessRebuild.php (hunk 6 above)
Comment (jbickar) [inline Stanford/Utility/Install/NodeAccessRebuild.php]: I learned it by watching you, alright!

File: Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php (hunk 1 above)
Comment (jbickar) [inline Stanford/JumpstartVPSA/Install/JSVPSAFinishInstallation.php]: BASIC-1920

**Comment by jbickar** [comment]
:+1:

**Pull request merged**